include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -67,60 +57,59 @@ struct FFT<2,false, Scalar, device_capabilities>
         exchangeValues<SharedMemoryAccessor, Scalar>(lo, hi, threadID, stride, sharedmemAdaptor);
         
         // Get twiddle with k = threadID mod stride, halfN = stride
-        fft::DIF<Scalar>::radix2(fft::twiddle<false, Scalar>(threadID & (stride - 1), stride), lo, hi);    
+        hlsl::fft::DIF<Scalar>::radix2(hlsl::fft::twiddle<false, Scalar>(threadID & (stride - 1), stride), lo, hi);    
     }
 
 
     template<typename Accessor, typename SharedMemoryAccessor>
     static void __call(NBL_REF_ARG(Accessor) accessor, NBL_REF_ARG(SharedMemoryAccessor) sharedmemAccessor)
     {
         // Set up the MemAdaptors
-        MemoryAdaptor<Accessor, _NBL_HLSL_WORKGROUP_SIZE_ << 1> memAdaptor;
+        MemoryAdaptor<Accessor, 1> memAdaptor;
         memAdaptor.accessor = accessor;
         MemoryAdaptor<SharedMemoryAccessor> sharedmemAdaptor;
         sharedmemAdaptor.accessor = sharedmemAccessor;
 
-        // Compute the SubgroupContiguousIndex only once
+        // Compute the indices only once
         const uint32_t threadID = uint32_t(SubgroupContiguousIndex());
+		const uint32_t loIx = threadID;
+		const uint32_t hiIx = loIx + _NBL_HLSL_WORKGROUP_SIZE_;
 
         // Read lo, hi values from global memory
         vector <Scalar, 2> loVec;
         vector <Scalar, 2> hiVec;
-        memAdaptor.get(threadID, loVec);
-        memAdaptor.get(threadID + _NBL_HLSL_WORKGROUP_SIZE_, hiVec);
+        // TODO: if we get rid of the Memory Adaptor on the accessor and require comples getters and setters, then no `2*`
+        memAdaptor.get(2 * loIx , loVec);
+        memAdaptor.get(2 * hiIx, hiVec);
```

Comment:
ok lets do it, rely that `Accessor` has `get(uint32_t index, NBL_REF(complex_t<Scalar>) val)` and `set` methods